### ISSUE TITLE
Skype agent "disconnects" until "waken up" by agent chatting with the bot

### ISSUE THREAD

**Issue by smartchatco** [tompaana/intermediator-bot-sample#33]
After much testing, on two different instances of this project, it seems at least to me that after listening on the skype channel for more than 24 hours, skype will temporarily disconnect if there is no messages sent from the skype agent to the bot.

So what happens is if there has been no activity, the agent is still listening on skype after a day, and then all of a sudden say someone on the webchat channel says "human" the agent waiting on skype will not get a notification.  I notice it does get added to the queue, but no notification gets through to skype. As soon as the agent happens to start messaging the bot it "reconnects" so that "human" requests will go through.  I am thinking this may be a skype channel specific thing perhaps a bot framework issue, but maybe it is this project not implementing a resume chat function correctly?

I have tried setting a routine that is triggered every 12 hours that messages from the bot to the Skype agent as a sort of workaround, but it does not seem to fix the issue.  It seems like it has to be from the agents side talking to the bot, to keep the Skype connection alive.  Perhaps someone can test and see if this is a bug or just me.

**Comment by tompaana** [comment]
@smartchatco Good catch, thanks! Would timeouts for requests and connections fix the issue? We could have something like `MessageRouterManager.CleanExpiredRequestsAndConnections()` method.

**Comment by tompaana** [comment]
There will be a massive overhaul to this project happening next week - this will be one issue that must be fixed.

**Comment by smartchatco** [comment]
I think what might be a good workaround is have all agents disconnect and pending requests cleared after 12 hours or 24 hours of inactivity by agents. Seems like a heavy lift though.  That way if the agents are disconnected they will be notified they are no longer receiving requests and can connect again.  When they connect again that will reset that timeout for the skype channel.

**Comment by tompaana** [comment]
Aye. We simply need to record/update the time of the last activity in respect to a connection. With requests the time the request was made should suffice.

**Comment by Lajeev** [comment]
@tompaana  I am having the same issue in slack channel, any update on this?
